src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -1371,26 +1371,24 @@ def get_run_step(self, step_id: UUID) -> StepRunModel:
             resource_model=StepRunModel,
         )
 
-    def get_run_step_outputs(self, step_id: UUID) -> Dict[str, ArtifactModel]:
-        """Get a list of outputs for a specific step.
+    def list_run_steps(
+        self, run_id: Optional[UUID] = None
+    ) -> List[StepRunModel]:
+        """Get all run steps.
 
         Args:
-            step_id: The id of the step to get outputs for.
+            run_id: If provided, only return steps for this pipeline run.
 
         Returns:
-            A dict mapping artifact names to the output artifacts for the step.
-
-        Raises:
-            ValueError: if the response from the API is not a dict.
+            A list of all run steps.
         """
-        body = self.get(f"{STEPS}/{str(step_id)}{OUTPUTS}")
-        if not isinstance(body, dict):
-            raise ValueError(
-                f"Bad API Response. Expected dict, got {type(body)}"
-            )
-        return {
-            name: ArtifactModel.parse_obj(entry) for name, entry in body.items()
-        }
+        filters = locals()
+        filters.pop("self")
+        return self._list_resources(
+            route=STEPS,
+            resource_model=StepRunModel,
+            **filters,
+        )
 
     def get_run_step_inputs(self, step_id: UUID) -> Dict[str, ArtifactModel]:
```

Comment:
Just a side-question: if `get_run_step_outputs` is replaced by `list_artifacts`, shouldn't `get_run_step_inputs` also be achievable through a `list_artifacts` call ?

Reply:
Unfortunately not since that information is not contained in the `ArtifactSchema`, see https://github.com/zenml-io/zenml/pull/977#discussion_r998077458